### PR TITLE
[CHK-2650] feat(wallet): add storage account + queues for pagoPA wallet

### PR DESCRIPTION
**Pull request by giovanniberti**
### List of changes

 * add storage account + queues for pagoPA wallet

### Motivation and context

These resources are needed to implement asynchronous update of wallet usages

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/domains/wallet-common/03_storage.tf [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+      enabled = true
+      days    = 3
+    }
+  }
+}
+
+locals {
+  queue_alert_props = var.env_short == "p" ? [
+    {
+      "queue_key"   = "transactions-expiration-queue"
```
Comment (pietro-tota) [inline src/domains/wallet-common/03_storage.tf:133]:
```suggestion
      "queue_key"   = "wallet-usage-update-queue"
```

Comment (giovanniberti) [reply src/domains/wallet-common/03_storage.tf:133]: 🤦, fixing!

File: src/domains/wallet-common/env/weu-prod/terraform.tfvars [hunk 2]
```diff
@@ -75,3 +76,13 @@ redis_wallet_params = {
...
 
 enable_iac_pipeline = true
+
+wallet_storage_params = {
+  enabled                       = true
+  tier                          = "Standard"
+  kind                          = "StorageV2"
+  account_replication_type      = "GZRS",
+  advanced_threat_protection    = true,
+  retention_days                = 30,
```
Comment (pietro-tota) [inline src/domains/wallet-common/env/weu-prod/terraform.tfvars:86]: Are 30 retention days a valid values for wallet use case or maybe can be lowered to 7?

**Review by pietro-tota — COMMENTED** [review]
LGTM, just few comments

File: src/domains/wallet-common/env/weu-prod/terraform.tfvars (hunk 2 above)
Comment (giovanniberti) [reply src/domains/wallet-common/env/weu-prod/terraform.tfvars:86]: Yeah, maybe even 3 could be enough, for now I'll lower to 7

Comment (giovanniberti) [reply src/domains/wallet-common/env/weu-prod/terraform.tfvars]: Changed in 3abd91c4

**Review by pietro-tota — DISMISSED** [review]

File: src/domains/wallet-common/03_storage.tf [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    default_action             = "Deny"
+    ip_rules                   = []
+    virtual_network_subnet_ids = [module.wallet_storage_snet.id]
+    bypass                     = ["AzureServices"]
+  } : null
+  tags = var.tags
+}
+
+resource "azurerm_storage_queue" "wallet_usage_update_queue" {
+  name                 = "${local.project}-wallet-usage-update-queue"
```
Comment (infantesimone) [inline src/domains/wallet-common/03_storage.tf:74]: `wallet` is already present in `local.project` (`pagopa-d/u/p-weu-wallet`), so you might think about removing it.
looking terraform plan the name of this resource would become `pagopa-d/u/p-weu-wallet-wallet-usage-update-queue`

**Review by infantesimone — CHANGES_REQUESTED** [review]

File: src/domains/wallet-common/03_storage.tf (hunk 3 above)
Comment (giovanniberti) [reply src/domains/wallet-common/03_storage.tf]: Fixed in 444830f2

**Review by infantesimone — COMMENTED** [review]
The PR sounds good, just some considerations on the names of resources that could have double `wallet`. We could rename the resources as `payment-wallets `rather than `wallets`, to distinguish this wallet from the other wallets in PagoPA.
p.s maybe renaming project name in a dedicated PR

**Review by infantesimone — APPROVED** [review]

**Pull request merged**